metals/src/main/scala/scala/meta/internal/metals/NewFilesProvider.scala

```diff
@@ -66,31 +135,80 @@ class NewFilesProvider(
       )
   }
 
+  private def createWorksheet(directory: Option[AbsolutePath], name: String)(
+      implicit ec: ExecutionContext
+  ): Future[AbsolutePath] = {
+    val path = directory.getOrElse(workspace).resolve(name + ".worksheet.sc")
+    createFile(path)
+  }
+
   private def createFile(
       path: AbsolutePath
-  )(implicit ec: ExecutionContext): Future[URI] = {
+  )(implicit ec: ExecutionContext): Future[AbsolutePath] = {
     val result = Future {
-      Files.createFile(path.toNIO).toUri()
+      AbsolutePath(
+        Files.createFile(path.toNIO)
+      )
     }
     result.onFailure {
-      case NonFatal(e) => scribe.error("Cannot create file", e)
+      case NonFatal(e) =>
+        scribe.error("Cannot create file", e)
+        client.showMessage(
+          MessageType.Error,
+          s"Cannot create file:\n ${e.toString()}"
+        )
     }
     result
   }
 
   private def createFileAndWriteText(path: AbsolutePath, text: String)(
       implicit ec: ExecutionContext
-  ): Future[URI] = {
-    createFile(path).map { newFileUri =>
+  ): Future[AbsolutePath] = {
+    createFile(path).map { _ =>
       path.writeText(text)
-      newFileUri
+      path
     }
   }
 
+  private def openFile(path: AbsolutePath): Unit = {
+    client.metalsExecuteClientCommand(
+      new ExecuteCommandParams(
+        ClientCommands.GotoLocation.id,
+        List(
+          new Location(path.toString(), new org.eclipse.lsp4j.Range()): Object
```

Comment:
`Range` fixed.
Good catch with `toURI`, thanks!
`:Object` can not be ommitted, since it's Java List eventually, and it's not covariant. It's the same with already existing cases of calling `ClientCommands.GotoLocation`